Rollbar/Deploys/DeployApiCallResult.h

```diff
@@ -1,25 +1,19 @@
-//
-//  ApiCallResult.h
-//  Rollbar
-//
-//  Created by Andrey Kornich (Wide Spectrum Computing LLC) on 2018-09-17.
 //  Copyright © 2018 Rollbar. All rights reserved.
-//
 
 #import <Foundation/Foundation.h>
 
 #import "DeploymentDetails.h"
-#import "../RollbarJSONFriendlyObject.h"
+#import "RollbarJSONFriendlyObject.h"
 
-typedef enum DeployApiCallOutcome {
-    Outcome_Success,
-    Outcome_Error,
-} DeployApiCallOutcome;
+typedef NS_ENUM(NSInteger, DeployApiCallOutcome) {
+    DeployApiCallSuccess,
+    DeployApiCallError,
+};
 
 @interface DeployApiCallResult : RollbarJSONFriendlyObject
 @property (readonly) DeployApiCallOutcome outcome;
 @property (readonly, copy) NSString *description;
-// Designated Initializer:
+
 - (id)initWithResponse:(NSHTTPURLResponse*)httpResponse
```

Comment:
btw, one more thing learned while looking up info on NS_ENUM:
`- (instancetype)init NS_DESIGNATED_INITIALIZER;`
in other words returning (instancetype) instead of (id) from any static/class method and marking a designated initializer with the NS_DESIGNATED_INITIALIZER macro suffix.
[https://developer.apple.com/library/archive/releasenotes/ObjectiveC/ModernizationObjC/AdoptingModernObjective-C/AdoptingModernObjective-C.html#//apple_ref/doc/uid/TP40014150-CH1-SW6](https://developer.apple.com/library/archive/releasenotes/ObjectiveC/ModernizationObjC/AdoptingModernObjective-C/AdoptingModernObjective-C.html#//apple_ref/doc/uid/TP40014150-CH1-SW6)
Do we want to use these?

Reply:
There is no downside to returning `instancetype` unless you really want to signal that you do not want the class to be subclassed. Marking with designated initializer is not really necessary. It depends on the design. Here it would probably just add noise.